### PR TITLE
We now test the release code at each commit.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
#1125 but better.

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
For windows and macos, I grouped the tests and building of the wheels in the same job to have a hot cache.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -156,6 +127,7 @@ jobs:
             build_pip_pkg
           bazel-bin/build_pip_pkg artifacts
       - uses: actions/upload-artifact@v1
+        if: github.event_name == 'release' || matrix.python-version == '3.5'
```
Comment (seanpmorgan) [inline .github/workflows/release.yml]: Wondering why we would need to upload-artifact for CI testing?

Comment (gabrieldemarmiesse) [reply .github/workflows/release.yml]: To test the small job where we gather actifacts. If you see a simpler way, I'm open to ideas.

Comment (seanpmorgan) [reply .github/workflows/release.yml]: Gotcha. Better to test that than not test I suppose, but I do think the most likely cause of a failure there would be timeout or something.

**Comment by gabrieldemarmiesse** [comment]
It's going to make it easier to switch the docker images :)

**Pull request merged**